src/findings_data_import.py

```diff
@@ -49,12 +53,372 @@ def move_processed_file(s3_client, bucket, folder, filename):
+        return temp_data_filepath, findings_data
+    except json.JSONDecodeError:
+        logging.error("Unable to decode JSON data for %s", data_filename)
+        raise
+    except ClientError:
+        logging.error("Error downloading file %s from %s", data_filename, s3_bucket)
+        raise
+
+
+def validate_v1_findings(findings_data: list, field_map_dict: dict):
+    """Validate a list of V1 findings, discarding invalid entries (such as those with no severity which are not explicitly 'findings').
+
+    Parameters
+    ----------
+    findings_data : list
+        The findings list pulled from a findings JSON file, v1-style
+
+    field_map_dict: dict
+        The dictionary of replacement rules for field names in findings_data (used only in V1 style processing)
+
+    Returns
+    -------
+    list : A list of findings objects from findings_data that pass validation
+    """
+    valid_findings = []
+    # Iterate through data and save each record to the database
+    for index, finding in enumerate(findings_data):
+
+        if not finding or not isinstance(finding, dict):
+            logging.warning("Received an empty or invalid finding object, skipping.")
+            continue
+
+        # Replace or rename fields according to the field mapping configuration
+        for field in field_map_dict:
+            if field in finding.keys():
+                if field_map_dict[field]:
+                    finding[field_map_dict[field]] = finding[field]
+                finding.pop(field, None)
+
+        # work with v1 and v2. If has NCATS ID OR findings the document is probably OK
+        if (
+            "RVA ID" not in finding.keys()
+            or "NCATS ID" not in finding.keys()
+            or "Severity" not in finding.keys()
+        ):
+            logging.warning(
+                'Skipping record %d. Missing "RVA ID", "NCATS ID", or "Severity" field.',
+                index,
+            )
+            continue
+
+        # Get RVA ID in format DDDD([.-]D+) from the end of the "RVA ID" field.
+        rvaId = re.search(r"(\d{4})(?:[.-](\d+))?$", finding["RVA ID"])
+        if rvaId:
+            rID = f"RV{rvaId.group(1)}"
+            if rvaId.group(2) is not None:
+                rID += f".{rvaId.group(2)}"
+            finding["RVA ID"] = rID
+        else:
+            logging.warning(
+                'Skipping record %d: Unable to extract valid RVA ID from "%s"',
+                index,
+                finding["RVA ID"],
+            )
+            continue
+        # flag this as V1 so update knows how to handle it, and its clear to folks viewing the data downstream
+        finding["Schema"] = V1_SCHEMA
+        valid_findings.append(finding)
+
+    return valid_findings
+
+
+def validate_v2_findings(findings_data: dict):
+    """Validate a list of V1 findings, discarding invalid entries (such as those with no severity which are not explicitly 'findings').
```

Comment:
Please update the comment to reflect validating the v2 findings vs the v1 findings

Reply:
Comment was updated in [26198f1](https://github.com/cisagov/findings-data-import-lambda/pull/55/commits/26198f1b8328c746c1d0fa8245eb204adb3280ec), this conversation can be closed @jsf9k